enhancements/subscription-content/subscription-content-access.md

```diff
@@ -0,0 +1,242 @@
+---
+title: subscription-content-access
+authors:
+  - @adambkaplan
+reviewers:
+  - @gabemontero
+  - @sbose
+  - @abhgupta
+  - @luciddreamz
+  - @sideangleside
+  - Barnaby Court
+  - Chris Snyder
+approvers:
+  - @bparees
+  - @derekwaynecarr
+creation-date: 2019-06-19
+last-updated: 2019-06-19
+Status: implementable
+see-also:
+  - /enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md
+replaces:
+  - https://github.com/openshift/enhancements/pull/214
+superseded-by:
+---
+
+# Subscription Content Access
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in
+      [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+## Summary
+
+Accessing RHEL subscription content in workloads is a challenge on OpenShift 4.x. Containers can
+download RHEL content via `yum install` if their hosts have subscriptions attached. In 4.x our
+preferred host OS (RHCOS) is not capable of attaching subscriptions individually. This enhancement
+proposal aims to restore the ability for containers to download RHEL content with minimal action
+from the developer or cluster admin.
+
+## Motivation
+
+Containers can download RHEL content if their hosts have subscriptions attached. In 3.x, all nodes
+were assumed to be RHEL 7 nodes that were subscribed individually. The information needed to access
+subscription content (entitlement keys and redhat.repo configurations) was symbolically linked into
+the running container at a well-known location that yum/dnf could find. This capability existed in
+our patch of Docker, and was carried over into Red Hat’s container toolchain used on OpenShift 4
+(cri-o, podman, buildah, etc.).
+
+In OpenShift 4 the preferred operating system (RHCOS) is not capable of attaching subscriptions
+individually. This means that by default containers running on OpenShift 4 cannot download RHEL
+content. There are at least two use cases where this capability is relevant:
+
+Builds which download RHEL libraries - particularly those that are not available to the UBI images
+Kernel modules which need access to RHEL developer libraries.
+
+### Goals
+
+Identify the requirements needed to access subscription content on OpenShift. Describe the
+architectural components needed to provide seamless access to subscription content on OpenShift.
+Identify dependencies on non-OpenShift components that are needed to make goal #2 achievable.
+Provide a rough roadmap for implementation.
+
+### Non-Goals
+
+Describe the detailed implementation of the components described in the architecture. This will be
+accomplished in subsequent enhancement proposals.
+
+## Proposal
+
+### User Stories
```

Comment:
During the review of https://github.com/openshift/enhancements/pull/214 @derekwaynecarr explicitly asked for a user story around multi-cluster install.  And as a result, we landed with this user story:

```
As a multi-cluster administrator, I can enable my cluster for entitlements via a declarative API that works well 
with GitOps (or similar patterns) so that enabling entitled builds is not a burden as I scale the number of clusters 
under management.
```

Ideally, we either 
- carry that user story over here
- get confirmation from @derekwaynecarr such a story is not longer applicable here

Reply:
Will add in the next push